jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererRecommendationManager.kt

```diff
@@ -137,22 +137,23 @@ class CodeWhispererRecommendationManager {
         val caret = requestContext.editor.caretModel.primaryCaret
         val rightContext = document.charsSequence.subSequence(caret.offset, document.charsSequence.length).toString()
         val recommendationContent = recommendation.content()
-        val rightContextFirstLine = rightContext.substringBefore("\n")
-        val overlap =
-            if (recommendationContent.none { it == '\n' }) {
-                overlap(recommendationContent, rightContextFirstLine)
-            } else if (rightContextFirstLine.isEmpty()) {
-                overlap(recommendationContent, rightContext)
+
+        // we will ignore white space and new lines when matching right context
+        val overlap = overlap(recommendationContent.trim(), rightContext.trim())
+        val overlapIndex = recommendationContent.lastIndexOf(overlap)
+        val truncated = if (overlapIndex >= 0) {
+            // if the truncated recommendation only contains white space we will return an empty string instead
+            if (recommendationContent.substring(0, overlapIndex).trim().isNotEmpty()) {
+                recommendationContent.substring(0, overlapIndex)
             } else {
-                val tempOverlap = overlap(recommendationContent, rightContext)
-                if (recommendationContent.substring(0, recommendationContent.length - tempOverlap.length).none { it == '\n' }) {
-                    tempOverlap
-                } else {
-                    ""
-                }
```

Comment:
actually the old logic can not be simply removed. This if-else logic handles cases like if the first line is not empty, then our right-context algorithm can't simply apply since it's wrong to match a suffix of a multi-line completion to the prefix of the right context when the first line of right context is not empty. There will be display issues.

Reply:
The logic should simply be
1. get the overlap with the trimmed completion and right context, using the old if-else logic.
2. get the overlapIndex using lastIndexOf()
3. simply return this truncated one `recommendationContent.substring(0, overlapIndex)`. The last check is not needed since it's checked in the caller.

Reply:
added back check for first line of right context